server.go

```diff
@@ -547,7 +547,7 @@ func (c *serverConn) run(sctx context.Context) {
 			// branch. Basically, it means that we are no longer receiving
 			// requests due to a terminal error.
 			recvErr = nil // connection is now "closing"
-			if err == io.EOF || err == io.ErrUnexpectedEOF || errors.Is(err, syscall.ECONNRESET) {
+			if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) || errors.Is(err, syscall.ECONNRESET) {
 				// The client went away and we should stop processing
 				// requests, so that the client connection is closed
```

Comment:
Any opinions if we should log the error here? Perhaps debug level logging so we can provide some visibility into the error occurrence.

Reply:
I think these are in the category of expected errors and might be chatty. We could consider having a single debug log for the connection shutting down, whether through these errors or a shutdown. Doesn't need to be in this PR.